packages/elements-dev-portal/src/containers/StoplightProject.tsx

```diff
@@ -1,110 +1,116 @@
+import { RoutingProps } from '@stoplight/elements-core/types';
 import { pipe } from 'lodash/fp';
 import * as React from 'react';
-import { Redirect, useLocation } from 'react-router-dom';
+import { Link, Redirect, Route, useHistory, useParams } from 'react-router-dom';
+
+import { BranchSelector } from '../components/BranchSelector';
+import { DevPortalProvider } from '../components/DevPortalProvider';
+import { Loading } from '../components/Loading';
+import { NodeContent } from '../components/NodeContent';
+import { NotFound } from '../components/NotFound';
+import { TableOfContents } from '../components/TableOfContents';
+import { useGetBranches } from '../hooks/useGetBranches';
+import { useGetNodeContent } from '../hooks/useGetNodeContent';
+import { useGetTableOfContents } from '../hooks/useGetTableOfContents';
 
 export interface StoplightProjectProps extends RoutingProps {
-  /**
-   * The slug of your Stoplight Workspace.
-   * (Usually you can find this as part of your Stoplight URL: https://[workspaceSlug].stoplight.io)
-   */
-  workspaceSlug: string;
   /**
    * The slug of the Stoplight Project.
    * @example "elements"
    */
-  projectSlug: string;
+  projectId: string;
   /**
    * If your company runs an on-premise deployment of Stoplight,
    * set this prop to point the StoplightProject component at the URL of that instance.
    */
   platformUrl?: string;
-  /**
-   * The name of a specific branch of the project to show. If no branch is provided, the default branch will be shown.
-   */
-  branchSlug?: string;
-  /**
-   * *TBD*
-   */
-  authToken?: string;
 }
 
-const StoplightProjectImpl: React.FC<StoplightProjectProps> = ({
-  workspaceSlug,
-  platformUrl,
-  projectSlug,
-  branchSlug,
-  authToken,
-}) => {
-  const [firstItem, setFirstItem] = React.useState<Item>();
-  const { pathname } = useLocation();
+const StoplightProjectImpl: React.FC<StoplightProjectProps> = ({ projectId }) => {
+  const { branchSlug = '', nodeSlug = '' } = useParams<{ branchSlug?: string; nodeSlug: string }>();
+  const history = useHistory();
+
+  const { data: tableOfContents, isFetched: isTocFetched } = useGetTableOfContents({ projectId, branchSlug });
+  const { data: branches } = useGetBranches({ projectId });
+  const { data: node, isFetched } = useGetNodeContent({ nodeSlug, projectId, branchSlug });
 
-  if (pathname === '/' && firstItem) {
-    return <Redirect to={firstItem.uri} />;
+  let elem = <Loading />;
+  if (!nodeSlug) {
+    if (isTocFetched) {
+      if (tableOfContents?.items) {
+        const firstNode = findFirstNode(tableOfContents.items);
+        if (firstNode) {
+          return <Redirect to={branchSlug ? `/branches/${branchSlug}/${firstNode.slug}` : `/${firstNode.slug}`} />;
+        }
+      }
+    }
+  } else if (!isFetched) {
+    elem = <Loading />;
+  } else if (!node) {
+    elem = <NotFound />;
+  } else {
+    elem = <NodeContent node={node} Link={Link} />;
```

Comment:
This logic is a bit convoluted. What about rewriting it like so:
```tsx
if(!nodeSlug && .... && ) {
  // ...
  return <Redicrect
}
let elem = <NotFound />;
if(node){
  elem = <NodeContent .../>
}
else if(...){
  elem = <Loading />
}

```

Reply:
Could you push a commit for any code changes you recommend?